### ISSUE TITLE
How do I get the IP address of a client that makes a request to the server?

### ISSUE THREAD

**Issue by heitoke** [elysiajs/elysia#145]
Yes, I saw this issue(#106), but it didn't help me, so I ask the question again

**Comment by Xavi007** [comment]
To get the IP address of a client that makes a request to the server in ElysiaJS, you can use the following code:

JavaScript
"""
import { Elysia } from 'elysia';

const elysia = new Elysia();

elysia.on('request', (req, res) => {
  // Get the client IP address from the `req.headers` object.
  const clientIp = req.headers['X-Forwarded-For'] || req.headers['CF-Connecting-IP'] || req.socket.remoteAddress;

  // Use the client IP address in your application.
  console.log(`The client IP address is ${clientIp}`);
});

elysia.listen(3000);

"""
The req.headers object contains all of the HTTP headers that were sent in the request. The X-Forwarded-For header is typically set by a proxy server, such as Cloudflare or Nginx. If the request is not coming from a proxy server, then the req.socket.remoteAddress property will contain the IP address of the client.

Example:
"""
JavaScript
import { Elysia, rateLimit } from 'elysia';

const elysia = new Elysia();

elysia.use(rateLimit({
  keyGenerator: (req) => req.headers['X-Forwarded-For'] || req.headers['CF-Connecting-IP'] || req.socket.remoteAddress,
  max: 10,
  duration: 60000,
}));

elysia.on('request', (req, res) => {
  // ...
});

elysia.listen(3000);

"""

This code will limit each client to 10 requests per minute. If a client makes more than 10 requests per minute, they will be blocked.

Note: If you are using ElysiaJS with a proxy server, you may need to configure the proxy server to forward the client IP address to your ElysiaJS server.

Additional notes:

It is important to note that getting the client IP address is not always reliable. For example, if the client is using a VPN or a proxy server, then you may not be able to get the real IP address of the client.
If you need to get the real IP address of the client, then you may need to use a third-party service such as IP Geolocation API.
Conclusion:

Getting the client IP address in ElysiaJS is a straightforward process. You can use the req.headers object or the req.socket.remoteAddress property to get the client IP address. However, it is important to note that getting the client IP address is not always reliable.

**Comment by heitoke** [comment]
Yes, thank you, it helped me, but I'm not writing in JS, but in TS.

Because of this, I don't have a `request.socket.remoteAddress'. And I also don't have an ip if I use `localhost:3000` or `192.168.0.105:3000` ((

But I tried to make a request through `ngrok` and everything worked out)) Which means it will work in production.

**Comment by Xavi007** [comment]
As you mentioned, the req.socket.remoteAddress property is not available in TypeScript. However, you can still get the client IP address by using the req.headers object.

If you are using ElysiaJS with a proxy server, you may need to configure the proxy server to forward the client IP address to your ElysiaJS server.

I am glad to hear that you were able to get the client IP address to work in production using ngrok.

**Comment by chefnaphtha** [comment]
nigga thought he was slick using chatgpt
